src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs

```diff
@@ -349,5 +394,179 @@ private static List<OtlpCommon.KeyValue> ToOtlpAttributes(KeyValuePair<string, o
 
             return attributes;
         }
+
+        private static Action<RepeatedField<OtlpTrace.Span>, int> CreateRepeatedFieldOfSpanSetCountAction()
+        {
+            FieldInfo repeatedFieldOfSpanCountField = typeof(RepeatedField<OtlpTrace.Span>).GetField("count", BindingFlags.NonPublic | BindingFlags.Instance);
+
+            DynamicMethod dynamicMethod = new DynamicMethod(
+                "CreateSetCountAction",
+                null,
+                new[] { typeof(RepeatedField<OtlpTrace.Span>), typeof(int) },
+                typeof(ActivityExtensions).Module,
+                skipVisibility: true);
+
+            var generator = dynamicMethod.GetILGenerator();
```

Comment:
Not blocking this PR: .NET seems to spend lots of energy recently for AOT (ahead of time) compilation and the direction to move to [yet another macro / mata-programming](https://devblogs.microsoft.com/dotnet/introducing-c-source-generators/) instead of emitting IL at runtime.